components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java

```diff
@@ -368,10 +368,20 @@ private Map<String, Object> getUserClaimsInOIDCDialect(OAuthAuthzReqMessageConte
         if (isEmpty(userAttributes)) {
             if (isLocalUser(authzReqMessageContext)) {
                 if (log.isDebugEnabled()) {
-                    log.debug("User attributes not found in cache. Trying to retrieve attribute for local user: " +
-                            authzReqMessageContext.getAuthorizationReqDTO().getUser());
+                    log.debug("User attributes not found in cache. Trying to retrieve attribute from auth " +
+                            "context for local user: " + authzReqMessageContext.getAuthorizationReqDTO().getUser());
+                }
+                userAttributes = authzReqMessageContext.getAuthorizationReqDTO().getUser()
```

Comment:
can we check the duplicated code between getUserClaimsInOIDCDialect(OAuthAuthzReqMessageContext authzReqMessageContext) and getUserClaimsInOIDCDialect(TokenRequestContext ) and merge them? Possible merge can be done here https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/17410a57cce99b6a9899db3d84acfa7f45e13033/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java#L403